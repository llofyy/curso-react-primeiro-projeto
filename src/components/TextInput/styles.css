.textInput {
    width: 100%;
    height: 40px;

    padding: 20px;
    font-size: 19px;

    margin-bottom: 30px;
}