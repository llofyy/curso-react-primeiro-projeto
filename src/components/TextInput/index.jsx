import './styles.css';
import Types from 'prop-types';

function TextInput({ value, onChange }) {
  return <input className="textInput" placeholder="FaÃ§a uma busca.." onChange={onChange} value={value} type="search" />;
}

TextInput.propTypes = {
  value: Types.string.isRequired,
  onChange: Types.func.isRequired,
};

export default TextInput;
