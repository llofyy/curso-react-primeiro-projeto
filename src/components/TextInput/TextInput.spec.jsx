import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TextInput from '.';

describe('<TextInput />', () => {
  it('should render a input with a onchange function', () => {
    const fn = jest.fn();
    render(<TextInput onChange={fn} value={'testando'} />);
    const input = screen.getByPlaceholderText('Faça uma busca..');
    userEvent.type(input, 'Test');
    expect(fn).toHaveBeenCalledTimes(4);
  });

  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput onChange={fn} value={'testando'} />);

    const input = screen.getByPlaceholderText(/Faça uma busca../i);
    expect(input.value).toBe('testando');
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput onChange={fn} value={'testando'} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
