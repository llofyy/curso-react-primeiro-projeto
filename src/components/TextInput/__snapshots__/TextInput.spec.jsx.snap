// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="textInput"
  placeholder="FaÃ§a uma busca.."
  type="search"
  value="testando"
/>
`;
